backend/eventloop/application_event_loop/application_event_runner_syncruns.go

```diff
@@ -33,20 +33,93 @@ func (action *applicationEventLoopRunner_Action) applicationEventRunner_handleSy
 	// Handle all GitOpsDeploymentSyncRun related events
 	err := action.applicationEventRunner_handleSyncRunModifiedInternal(ctx, dbQueries)
 
-	// TODO: GITOPSRVCE-44: Get the SyncRunModified object from k8s, so we can update it if necessary
-
-	// TODO: GITOPSRVCE-44: Add the ErrorOccurred condition to the syncRunModified
-
-	// TODO: GITOPSRVCE-44: If no user error (just dev error), then output generic error occurred
+	syncRunCR, clientErr := getGitOpsDeploymentSyncRun(ctx, action.workspaceClient, action.eventResourceName, action.eventResourceNamespace)
+	if clientErr != nil {
+		if !apierr.IsNotFound(clientErr) {
+			return fmt.Errorf("unable to get GitOpsDeploymentSyncRun: %v", err)
+		}
+		return nil
+	}
 
+	conditionType := managedgitopsv1alpha1.GitOpsDeploymentSyncRunConditionErrorOccurred
 	if err != nil {
+		// set the condition only for user errors
+		if err.UserError() != "" {
```

Comment:
I don't understand thing. Can you explain what you are trying to achieve?

Reply:
The `err` here contains both user-focussed error and developer-focussed error messages. We want only user errors to be reported in the status and log dev-related errors.